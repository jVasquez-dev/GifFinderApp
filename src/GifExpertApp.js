import React, {useState} from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

export const GiftExpertApp = () => {

    // const categories = ['Pluto', 'JJK', 'Bleach'];
    const [categories, setCategories] = useState(['Naruto'])

    // const handleAdd = () => {
    //     setCategories([...categories, 'Naruto']);
    //     setCategories( cats => [...categories, 'Naruto'] );
    // } 

    return (
        <>
        <h2>GiftExpertApp</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr />

        <ol>
            {
                categories.map( category => (
                    <GiftGrid 
                        key={ category }
                        category={ category }
                    />

                ))
            }
        </ol>
        </>
    )
}
