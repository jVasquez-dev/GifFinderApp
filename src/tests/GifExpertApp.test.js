import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('<GiftExpertApp /> tests', () => {


    test('should render correctly', () => {
        
        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();

    })

    test('should shows categories', () => {
        
        const categories = ['Metallica', 'Slipknot']
        const wrapper = shallow( <GifExpertApp defaultCategories={categories}/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    })
    
    
})