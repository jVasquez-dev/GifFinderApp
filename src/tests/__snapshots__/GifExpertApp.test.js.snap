// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftExpertApp /> tests should render correctly 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`<GiftExpertApp /> tests should shows categories 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Metallica"
      key="Metallica"
    />
    <GifGrid
      category="Slipknot"
      key="Slipknot"
    />
  </ol>
</Fragment>
`;
